### PR TITLE
Update iframe-transport-vendors.js to use new experimental b4a library

### PR DESCRIPTION
**Pull request by sushan04**
# Instructions:

- Pick a meaningful title for your pull request. (Use sentence case.)
  - Prefix the title with an emoji to identify what is being done. (Copy-paste the emoji itself (not the :code:) from the list below.)
  - Do not overuse punctuation in the title (like `(chore):`).
  - If it is helpful, use a simple prefix (like `ProjectX: Implement some feature`).
- Enter a succinct description that says why the PR is necessary, and what it does.
  - Mention the GitHub issue that is being addressed by the pull request.
  - The keywords `Fixes`, `Closes`, or `Resolves` followed the issue number will automatically close the issue.

# Example of a good description:

- Implement aspect X
- Leave out feature Y because of A
- Improve performance by B
- Improve accessibility by C

# Emojis for categorizing pull requests:

✨ New feature (`:sparkles:`)  
🐛 Bug fix (`:bug:`)  
🔥 P0 fix (`:fire:`)  
✅ Tests (`:white_check_mark:`)  
🚀 Performance improvements (`:rocket:`)  
🖍 CSS / Styling (`:crayon:`)  
♿ Accessibility (`:wheelchair:`)  
🌐 Internationalization (`:globe_with_meridians:`)  
📖 Documentation (`:book:`)  
🏗 Infrastructure / Tooling / Builds / CI (`:building_construction:`)  
⏪ Reverting a previous change (`:rewind:`)  
♻️ Refactoring (like moving around code w/o any changes) (`:recycle:`)  
🚮 Deleting code (`:put_litter_in_its_place:`)

### CODE REVIEW

File: extensions/amp-analytics/0.1/iframe-transport-vendors.js [hunk 1]
```diff
@@ -25,6 +25,6 @@
  * @const {!JsonObject}
  */
 export const IFRAME_TRANSPORTS = /** @type {!JsonObject} */ ({
-  'bg': 'https://tpc.googlesyndication.com/b4a/b4a-runner.html',
+  'bg': 'https://tpc.googlesyndication.com/b4a/experimental/b4a-runner.html',
```
Comment (lannka) [inline extensions/amp-analytics/0.1/iframe-transport-vendors.js]: as discussed offline, could you add it as a new config block

```js
export const IFRAME_TRANSPORTS_CANARY = {
  'bg': 'https://tpc.googlesyndication.com/b4a/experimental/b4a-runner.html',
  'moat': 'https://z.moatads.com/ampanalytics093284/iframe.html',
}
```

then you will change `vendors.js`

```js
mergeIframeTransportConfig(ANALYTICS_CONFIG, isCanary(self) ? IFRAME_TRANSPORTS_CANARY : IFRAME_TRANSPORTS);

```

**Comment by lannka** [comment]
also pls update PR description

**Comment by sushan04** [comment]
This PR creates a separate configuration to use when creating iframe transports in canary for approved vendors. This way the vendors can independently test a new iframe source in canary without affecting production traffic.

File: extensions/amp-analytics/0.1/iframe-transport-vendors.js (hunk 1 above)
Comment (sushan04) [reply extensions/amp-analytics/0.1/iframe-transport-vendors.js]: Done.

File: extensions/amp-analytics/0.1/vendors.js [hunk 2]
```diff
@@ -264,7 +264,7 @@ ANALYTICS_CONFIG['adobeanalytics_nativeConfig']
 ANALYTICS_CONFIG['oewa']['triggers']['pageview']['iframe' +
 /* TEMPORARY EXCEPTION */ 'Ping'] = true;
 
-mergeIframeTransportConfig(ANALYTICS_CONFIG, IFRAME_TRANSPORTS);
+mergeIframeTransportConfig(ANALYTICS_CONFIG, isCanary(self) ? IFRAME_TRANSPORTS_CANARY : IFRAME_TRANSPORTS);
```
Comment (lannka) [inline extensions/amp-analytics/0.1/vendors.js]: missing `import` of `isCanary`

**Comment by lannka** [comment]
you also need to sign the CLA

**Comment by sushan04** [comment]
I signed it!

File: extensions/amp-analytics/0.1/vendors.js (hunk 2 above)
Comment (sushan04) [reply extensions/amp-analytics/0.1/vendors.js]: Done.

Comment (sushan04) [reply extensions/amp-analytics/0.1/vendors.js]: Done.

**Comment by lannka** [comment]
obsolete.
see #22296

**Pull request closed without merging**